x-pack/plugins/security/server/session_management/session.ts

```diff
@@ -150,7 +151,7 @@ export class Session {
   async get(request: KibanaRequest) {
```

Comment:
My alternative approach is a sort of hybrid between what you have here, and what Oleg suggested with `Either`. My approach isn't better or worse, but is maybe more inspired by golang / c#. The c# variant would be more of a `tryGet` than a `get`, we're not required to adopt their naming convention ([example](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=net-6.0#examples))

We could have `get` return an object that contains an error or session, with a type definition similar to:

```ts
type GetSessionResult = { error: SessionError; value: null } | { error: null; value: SessionValue };
```

This would allow you to destructure this in your calling code somewhat cleanly, and you could avoid the `typeof`/`instanceof` checks.

Reply:
Thanks for taking a look at this.

As I read your suggestion, you're actually suggesting two different approaches:

- Have a `get` function that throws the error and a `tryGet` wrapper that returns `null` incase of a thrown error (Oleg actually also suggested this to me IRL).
- Change the return type of `get` to be a object like `{ error: x, value: y }`.

The latter approach is more or less similar to "Either", but without all the functional programming religion and associated boilerplate.

For reference I pushed my Either implementation of this PR to a separate branch so you can check it out. Here's the commit that switches to use an Either monad approach: https://github.com/watson/kibana/commit/71a13c8854ee9aaeab0d48ad5699a6397d3cf134

Reply:
> As I read your suggestion, you're actually suggesting two different approaches

Ah, I was only proposing the second of the two (`Change the return type of get to be a object like { error: x, value: y }.`). I had a confusing preamble that made it seem otherwise. Sorry for the confusion.


> The latter approach is more or less similar to "Either", but without all the functional programming religion and associated boilerplate.

++ yep, that was my thinking.


> For reference I pushed my Either implementation of this PR to a separate branch so you can check it out. Here's the commit that switches to use an Either monad approach: https://github.com/watson/kibana/commit/71a13c8854ee9aaeab0d48ad5699a6397d3cf134

Thanks for sharing. My lack of experience with FP definitely biases my perspective here, but I don't think that does much to improve readability in a codebase that is primarily not in the FP style. That said, I think it was a worthwhile exercise for you to do, and I wouldn't personally block this PR if you felt that was the best approach.